Sources/Sentry/SentryTracer.m

```diff
@@ -140,24 +141,30 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
         // Store current amount of frames at the beginning to be able to calculate the amount of
         // frames at the end of the transaction.
         SentryFramesTracker *framesTracker = [SentryFramesTracker sharedInstance];
+#    if SENTRY_TARGET_PROFILING_SUPPORTED
+        framesTracker.currentTracer = self;
+#    endif // SENTRY_TARGET_PROFILING_SUPPORTED
         if (framesTracker.isRunning) {
             SentryScreenFrames *currentFrames = framesTracker.currentFrames;
             initTotalFrames = currentFrames.total;
             initSlowFrames = currentFrames.slow;
             initFrozenFrames = currentFrames.frozen;
         }
-#endif
+#endif // SENTRY_HAS_UIKIT
 #if SENTRY_TARGET_PROFILING_SUPPORTED
         if ([_hub getClient].options.enableProfiling) {
             [profilerLock lock];
             if (profiler == nil) {
                 profiler = [[SentryProfiler alloc] init];
                 [SentryLog logWithMessage:@"Starting profiler." andLevel:kSentryLevelDebug];
+#    if SENTRY_HAS_UIKIT
+                [framesTracker resetProfilingTimestamps];
+#    endif // SENTRY_HAS_UIKIT
```

Comment:
`m`: This is currently the only place we reset the profiling timestamps. If no new trace is started for a long time, the timestamps stay in memory for no reason.

Reply:
good catch, forgot to call it when compiling profile info for the end of a profiled transaction.